### PR TITLE
Unroll dict input before call Accelerator X_steps

### LINKED ISSUE #10907: Unroll dict input before call Accelerator X_steps and update function type
## Proposed refactor

From @justusschock's comments in #10890
Refactor accelerator  X_steps()'s signature, to support both positional and keyword arguments
Instead of unrolling step_kwarg dictionary in accelerator steps, unroll them in caller side. 

This will unblock #10648 move functions from accelerator to strategies part

### Motivation

Keep flexibilities and improving function typing

### Pitch

Current Accelerator X_step typing signature is 
https://github.com/four4fish/pytorch-lightning/blob/master/pytorch_lightning/accelerators/accelerator.py#L124-L148

Update to :
`def X_step(self, *args, **kwargs) -> Optional[STEP_OUTPUT]:`
Which not align with what torch model training step and model() signature.

In caller side, eg: optimizer_loop.py. Instead of passing dictionary `step_kwargs`, pass `*step_kwargs.values()`

### Additional context

______________________________________________________________________

#### If you enjoy Lightning, check out our other projects! ⚡

- [**Metrics**](https://github.com/PyTorchLightning/metrics): Machine learning metrics for distributed, scalable PyTorch applications.

- [**Lite**](https://pytorch-lightning.readthedocs.io/en/latest/starter/lightning_lite.html): enables pure PyTorch users to scale their existing code on any kind of device while retaining full control over their own loops and optimization logic.

- [**Flash**](https://github.com/PyTorchLightning/lightning-flash): The fastest way to get a Lightning baseline! A collection of tasks for fast prototyping, baselining, fine-tuning, and solving problems with deep learning.

- [**Bolts**](https://github.com/PyTorchLightning/lightning-bolts): Pretrained SOTA Deep Learning models, callbacks, and more for research and production with PyTorch Lightning and PyTorch.

- [**Lightning Transformers**](https://github.com/PyTorchLightning/lightning-transformers): Flexible interface for high-performance research using SOTA Transformers leveraging Pytorch Lightning, Transformers, and Hydra.

cc @justusschock @awaelchli @akihironitta

### PR DESCRIPTION
**Pull request by four4fish**
From @justusschock's comments in #10890
Refactor accelerator X_steps()'s signature, to support both positional and keyword arguments
Instead of unrolling step_kwarg dictionary in accelerator steps, unroll them in caller side.

This will unblock #10648 move functions from accelerator to strategies part
## What does this PR do?

Fixes #10907

## Before submitting

- [x] Was this **discussed/approved** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [x] Did you make sure to **update the documentation** with your changes? (if necessary)
- [ ] Did you write any **new necessary tests**? (not for typos and docs)
- [x] Did you verify new and **existing tests pass** locally with your changes?
- [x] Did you list all the **breaking changes** introduced by this pull request?
- [x] Did you **update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)**? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

- [x] Is this pull request ready for review? (if not, please submit in draft mode)
- [x] Check that all items from **Before submitting** are resolved
- [x] Make sure the title is self-explanatory and the description concisely explains the PR
- [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?

Make sure you had fun coding 🙃

### CODE REVIEW

**Review by ananthsub — APPROVED** [review]

**Review by tchaton — APPROVED** [review]
LGTM !

File: CHANGELOG.md [hunk 1]
```diff
@@ -86,6 +86,9 @@ The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/).
 - The `LightningModule.{add_to_queue,get_from_queue}` hooks no longer get a `torch.multiprocessing.SimpleQueue` and instead receive a list based queue ([#10034](https://github.com/PyTorchLightning/pytorch-lightning/pull/10034))
 
 
+- Changed `training_step`, `validation_step`, `test_step` and `predict_step` functions signatures in `Accelerator` and update input from caller side ([#10908](https://github.com/PyTorchLightning/pytorch-lightning/pull/10908))
```
Comment (awaelchli) [inline CHANGELOG.md:89]:
```suggestion
- Changed `training_step`, `validation_step`, `test_step` and `predict_step` method signatures in `Accelerator` and updated input from caller side ([#10908](https://github.com/PyTorchLightning/pytorch-lightning/pull/10908))
```

**Review by awaelchli — APPROVED** [review]

**Review by justusschock — APPROVED** [review]

File: pytorch_lightning/loops/epoch/evaluation_epoch_loop.py [hunk 2]
```diff
@@ -225,11 +225,11 @@ def _evaluation_step(self, batch: Any, batch_idx: int, dataloader_idx: int) -> O
         if self.trainer.testing:
             self.trainer.lightning_module._current_fx_name = "test_step"
             with self.trainer.profiler.profile("test_step"):
-                output = self.trainer.accelerator.test_step(step_kwargs)
+                output = self.trainer.accelerator.test_step(*step_kwargs.values())
```
Comment (carmocca) [inline pytorch_lightning/loops/epoch/evaluation_epoch_loop.py:228]: shouldn't this be `**step_kargs`?

Comment (awaelchli) [reply pytorch_lightning/loops/epoch/evaluation_epoch_loop.py:228]: This works because `values()` is an iterable that returns the pairs, right?

```
def foo(a=1, b=2):
    print(a, b)
    
args = dict(a=3, b=4)
foo(*args.values())
3 4
```


But `**` wouldn't work on that.

Comment (justusschock) [reply pytorch_lightning/loops/epoch/evaluation_epoch_loop.py:228]: @awaelchli not on the `values` but on the dict itself. The issue is that the order of `values` is not guaranteed to be consistent with the order of adding them to the dict for python<3.7 and even then we can't be sure we always add them in the correct order. So probably **step_kwargs is a bit safer as it unrolls them to keyword arguments (where the order does not matter). 

@carmocca I agree, however with the `*step_kwargs.values()` it is the way we currently do it.

Comment (carmocca) [reply pytorch_lightning/loops/epoch/evaluation_epoch_loop.py:228]: It's not about order because we use OrderedDict

It's actually because we support different names for the arguments, see they NEED to be passed positionally, e.g:

`def test_step(self, b, nb, dl_idx)`
